### PR TITLE
Benchmark encointer plallets

### LINKED ISSUE #21: benchmark encointer-parachain-runtime
depends on: https://github.com/encointer/pallets/issues/56

check: https://github.com/integritee-network/parachain/commit/536374facdefe5b1fd8d72b0fdf0e03a4fde8700

### PR DESCRIPTION
**Pull request by pifragile**
* Closes #21

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,24 +125,19 @@ In `./scripts` we have two scripts for benchmarking the runtimes.
...
-**Note**: With the current substrate version, we brick the `frame_system::Weight` file because the `set_changes_trie_config`
-has been removed from the pallet, but it still exists in the trait definition. So after running the benchmark, we need 
-to add the function manually again.
-
 ### Adding new pallets to be benchmarked
 Every pallet with a `type WeightInfo` parameter in its config must be benchmarked.
 
 1. [Cargo.toml] add `<new_pallet>/runtime-benchmarks` in the `runtime-benchmarks` feature section.
-2. [runtime] add the new pallet to the `list_benchmark!` and `add_benchmark!` list.
+2. [runtime] Add the new pallet to be benchmarked to the `define_benchmarks!` macro in the runtime.
```
Comment (clangenb) [inline README.md:140]: Nice: 👍

**Review by clangenb — COMMENTED** [review]
Nice, looks very good. Just please add the weight info in the runtime for:

* `pallet_collective`
* `pallet_membership`

**Comment by clangenb** [comment]
I am talking about this: https://github.com/encointer/encointer-parachain/blob/b006c1e1df27c32e9a15843b6b4aa9ff292721c4/polkadot-parachains/encointer-runtime/src/lib.rs#L680

Please replace `type WeightInfo = ()` with `type WeightInfo = weights::pallet_collective::WeightInfo<Runtime>`

**Review by clangenb — APPROVED** [review]
LGTM.

File: polkadot-parachains/encointer-runtime/src/weights/pallet_encointer_ceremonies.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	// Storage: EncointerCeremonies NewbieCount (r:0 w:1)
+	// Storage: EncointerCeremonies ReputableCount (r:0 w:1)
+	// Storage: EncointerCeremonies AssignmentCounts (r:0 w:1)
+	// Storage: EncointerCeremonies BootstrapperCount (r:0 w:1)
+	// Storage: EncointerCeremonies ReputableIndex (r:0 w:1)
+	// Storage: EncointerCeremonies AttestationCount (r:0 w:1)
+	// Storage: EncointerCeremonies EndorseeCount (r:0 w:1)
+	// Storage: EncointerCeremonies EndorseesCount (r:0 w:1)
+	fn purge_community_ceremony() -> Weight {
+		(131_000_000 as Weight)
```
Comment (clangenb) [inline polkadot-parachains/encointer-runtime/src/weights/pallet_encointer_ceremonies.rs:147]: @brenzi I forgot about this one, do we want to update that too similar to the encointer-node?

https://github.com/encointer/encointer-node/commit/8ac46d1aec5ccace957e959e6a1f69f372f149eb

btw, what was your bad experience?

Comment (brenzi) [reply polkadot-parachains/encointer-runtime/src/weights/pallet_encointer_ceremonies.rs:147]: bad experience was that the block time went up to several minutes while batch-purging with 800 purge extrinsics in one block (which means weight calculation was terribly wrong. No big problem for a solochain. But a parachain is bricked if block processing takes too long. So I'd rather manualley set a very high weight before we have better benchmarking

Comment (brenzi) [reply polkadot-parachains/encointer-runtime/src/weights/pallet_encointer_ceremonies.rs:147]: just set it equal to the node: 10^10

**Pull request merged**